src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -0,0 +1,297 @@
+    public: transport::Node::Publisher pub;
+
+    /// \brief Name of the selected model
+    public: QString modelName;
+
+    /// \brief List of the name of the links in the selected model
+    public: QStringList linkNameList;
+
+    /// \brief Index of selected link in list
+    public: int linkIndex{-1};
+
+    /// \brief Entity of the currently selected Link
+    public: std::optional<Entity> selectedEntity;
+
+    /// \brief True if a new entity was selected
+    public: bool changedEntity{false};
+
+    /// \brief True if a new link was selected from the dropdown
+    public: bool changedIndex{false};
+
+    /// \brief Force to be applied
+    public: math::Vector3d force{0.0, 0.0, 0.0};
+
+    /// \brief Torque to be applied
+    public: math::Vector3d torque{0.0, 0.0, 0.0};
+  };
+}
+}
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+ApplyForceTorque::ApplyForceTorque()
+  : GuiSystem(), dataPtr(std::make_unique<ApplyForceTorquePrivate>())
+{
+}
+
+/////////////////////////////////////////////////
+ApplyForceTorque::~ApplyForceTorque() = default;
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::LoadConfig(const tinyxml2::XMLElement */*_pluginElem*/)
+{
+  if (this->title.empty())
+    this->title = "Apply force and torque";
+
+  // Create wrench publisher
+  auto worldNames = gz::gui::worldNames();
+  if (!worldNames.empty())
+  {
+    auto topic = transport::TopicUtils::AsValidTopic(
+      "/world/" + worldNames[0].toStdString() + "/wrench");
+    if (topic == "")
+    {
+      gzerr << "Unable to create publisher" << std::endl;
+      return;
+    }
+    this->dataPtr->pub =
+      this->dataPtr->node.Advertise<msgs::EntityWrench>(topic);
+    gzdbg << "Created publisher to " << topic << std::endl;
+  }
+
+  gz::gui::App()->findChild<gz::gui::MainWindow *>
+    ()->installEventFilter(this);
+  gz::gui::App()->findChild<gz::gui::MainWindow *>
+    ()->QuickWindow()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+bool ApplyForceTorque::eventFilter(QObject *_obj, QEvent *_event)
+{
+  if (_event->type() ==
+    gz::sim::gui::events::EntitiesSelected::kType)
+  {
+    gz::sim::gui::events::EntitiesSelected *_e =
+        static_cast<gz::sim::gui::events::EntitiesSelected*>(_event);
+    this->dataPtr->selectedEntity = _e->Data().front();
+    this->dataPtr->changedEntity = true;
+  }
+  else if (_event->type() ==
+    gz::sim::gui::events::DeselectAllEntities::kType)
+  {
+    this->dataPtr->selectedEntity.reset();
+    this->dataPtr->changedEntity = true;
+  }
+
+  return QObject::eventFilter(_obj, _event);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::Update(const UpdateInfo &/*_info*/,
+  EntityComponentManager &_ecm)
+{
+  if (this->dataPtr->changedEntity)
+  {
+    this->dataPtr->changedEntity = false;
+
+    this->dataPtr->modelName = "";
```

Comment:
```suggestion
    this->dataPtr->modelName{""};
```

Reply:
I don't think this works, since this is an assignment, not an initialization. I get a compilation error if I try this.